network/server.go

```diff
@@ -704,7 +708,14 @@ func (s *DefaultServer) ForgetPeer(peer peer.ID, reason string) {
 
 // DisconnectFromPeer disconnects the networking server from the specified peer
 func (s *DefaultServer) DisconnectFromPeer(peerID peer.ID, reason string) {
```

Comment:
I think it sounds more reasonable if it accept `context.Context` parameter.
All the other external methods in this module should use context to preserve its span tree, too.

Reply:
this is big change, need split